src/modules/dataFetching/components/GenericTableWithData.tsx

```diff
@@ -65,7 +65,7 @@ const GenericTableWithData = <
   defaultPageSize = DEFAULT_ROWS_PER_PAGE,
   columns,
   recordType,
-  fetchPolicy,
+  fetchPolicy = 'cache-and-network',
```

Comment:
It's intentional that `cache-first` is the default policy for tables. It makes the UI appear a lot faster. For example switching back and forth between the enrollment and assessment tab on the client dashboard, if we use `cache-and-network` the user has to wait for a loading state every time.

It does mean we need to evict the cache when creating a new record or deleting a record, which is kind of annoying, but IMO worth it for the user experience. (It looks like you already added those cache evictions).

Reply:
I see, my understanding is that the 'cache-and-network' strategy provides the cached data immediately (like cache-first) and then also freshens it from network. The idea being you get fast-loading page initially and fresh data eventually. 

Not sure why that's not working here, I can take a look at it or just revert to the old fetch policy. It could be some interaction with the queryStatus.loading flag that's blocking the UI even though it has data to show.

I mostly added it since a full page reload is painfully slow (my guess is that this is a vite/MUI [bundle-size](https://mui.com/material-ui/guides/minimizing-bundle-size/) issue but that's another can of worms)

Reply:
Follow-up, yeah it does look like an issue with how we use `status.loading` in GenericTableWithData, I have a fix but it probably should go in another PR (if it's even worth pursuing). I'll revert the change here

Reply:
Oh, interesting. I just misunderstood the loading state, I guess. A separate PR would be great, if its not too much trouble. Or, just an explanation and I can try it out (like do you use the networkStatus to determine which type of loading state it's in, or just only return the loading indicator if `loading && !data` ? I think we show a full loading state when you navigate between pages right now, too. (I'm definitely not attached to that though).